### PR TITLE
Gap limit integration tests of current behaviour

### PR DESCRIPTION
**Pull request by majikandy**
2 tests.

1.  Funds outside the gap limit not visible

2.  Generate addresses in advance, then funds at that same address are visible

### CODE REVIEW

File: src/Stratis.Bitcoin.IntegrationTests/Wallet/CustomisableAddressGapLimitBeyondBIP44.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+﻿using Xunit;
+// ReSharper disable once InconsistentNaming
+
+namespace Stratis.Bitcoin.IntegrationTests.Wallet
+{
+    public partial class Customisable_address_gap_limit_beyond_BIP44
```
Comment (bokobza) [inline src/Stratis.Bitcoin.IntegrationTests/Wallet/CustomisableAddressGapLimitBeyondBIP44.cs]: These tests are not really about the customizable gap limit but rather about 1. testing the gap and 2. transactions being found when the number of addresses goes beyond the gap limit.
If you intend to add the gap limit testing tests here, then the name is fine.

Comment (majikandy) [reply src/Stratis.Bitcoin.IntegrationTests/Wallet/CustomisableAddressGapLimitBeyondBIP44.cs]: Yes good point, I was thinking of the future - but you are right, right now it isn't - I will change the name

File: src/Stratis.Bitcoin.IntegrationTests/Wallet/WalletAddressGenerationAndFundsVisibility_Steps.cs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            this.sendingStratisBitcoinNode.FullNode.Network.Consensus.CoinbaseMaturity = coinbaseMaturity;
+            this.receivingStratisBitcoinNode.FullNode.Network.Consensus.CoinbaseMaturity = coinbaseMaturity;
+
+            this.sharedSteps.MineBlocks(coinbaseMaturity + 1, this.sendingStratisBitcoinNode, AccountZero, SendingWalletName, WalletPassword);
+        }
+
+        private void a_wallet_with_funds_at_index_20_which_is_beyond_gap_limit()
+        {
+            ExtPubKey xPublicKey = this.GetExtendedPublicKey(ReceivingNodeName);
+            var recipientAddressBeyondGapLimit = xPublicKey.Derive(new KeyPath($"0/{20}")).PubKey.GetAddress(Network.RegTest);
```
Comment (YakupIpek) [inline src/Stratis.Bitcoin.IntegrationTests/Wallet/WalletAddressGenerationAndFundsVisibility_Steps.cs]: String interpolation is not needed in here.

Comment (majikandy) [reply src/Stratis.Bitcoin.IntegrationTests/Wallet/WalletAddressGenerationAndFundsVisibility_Steps.cs]: good point, I'll correct that

**Pull request merged**